src/Sylius/Behat/Context/Api/Admin/ManagingShipmentsContext.php

```diff
@@ -0,0 +1,121 @@
+     * @Then I should see a single shipment in the list
+     */
+    public function iShouldSeeCountShipmentsInList(int $count = 1): void
+    {
+        Assert::same($this->client->countCollectionItems(), $count);
+    }
+
+    /**
+     * @Then I should see the shipment of order :orderNumber as :shippingState
+     */
+    public function iShouldSeeTheShipmentOfOrderAs(string $orderNumber, string $shippingState): void
+    {
+        Assert::true(
+            $this->isShipmentWithOrderNumberAndState($orderNumber, $shippingState),
+            sprintf('Shipment for order %s with state %s does not exist', $orderNumber, $shippingState)
+        );
+    }
+
+    /**
+     * @Then /^I should see shipment for (the "[^"]+" order) as (\d+)(?:|st|nd|rd|th) in the list$/
+     */
+    public function iShouldSeeShipmentForTheOrderInTheList(string $orderNumber, int $position): void
+    {
+
+        Assert::true(
+            $this->client->hasItemOnPositionWithValue(--$position, 'order', sprintf('/new-api/orders/%s', $orderNumber)),
+            sprintf('On position %s there is no shipment for order %s', $position, $orderNumber)
+        );
+    }
+
+    /**
+     * @Then the shipment of the :orderNumber order should be :shippingState for :customer
+     * @Then the shipment of the :orderNumber order should be :shippingState for :customer in :channel channel
+     */
+    public function shipmentOfOrderShouldBe(
+        string $orderNumber,
+        string $shippingState,
+        CustomerInterface $customer,
+        ChannelInterface $channel = null
+    ): void {
+        $this->client->index('shipments');
+
+        foreach ($this->client->getCollectionItemsWithValue('state', StringInflector::nameToLowercaseCode($shippingState)) as $shipment) {
+
+            $orderIri = $shipment['order'];
+            $this->client->showByIri($orderIri);
+
+            if (!$this->client->responseHasValue('number', $orderNumber)) {
+                throw new \InvalidArgumentException(sprintf('There is no shipment with order %s.', $orderNumber));
+            }
+
+            if (!$this->client->relatedResourceHasValue('customer', 'email', $customer->getEmail())) {
+                throw new \InvalidArgumentException(sprintf('There is no shipment with customer %s.', $customer->getEmail()));
+            }
+
+            $this->client->showByIri($orderIri);
+            if (!$this->client->relatedResourceHasValue('channel', 'name', $channel->getName())) {
+                throw new \InvalidArgumentException(sprintf( 'There is no shipment in channel %s', $channel->getName()));
+            }
+        }
+    }
+
+    private function isShipmentWithOrderNumberAndState(string $orderNumber, string $state): bool
+    {
+        $items = $this->client->getCollection();
+
+        foreach ($items as $item) {
+            if ($item['order'] === sprintf('/new-api/orders/%s', $orderNumber) && $item['state'] === strtolower($state)) {
```

Comment:
Maybe use here an `IriConverter` service to create an IRI for order in a proper way?

Reply:
`IriConverter` can convert object to iri, here i have only `orderNumber`, what i can do is transform `orderNumber` to object and next convert object to IRI I'm not sure if that makes sense.

Reply:
Associated with this comment: https://github.com/Sylius/Sylius/pull/11208#discussion_r391484176